src/diffusers/loaders/peft.py

```diff
@@ -56,6 +56,37 @@
 }
 
 
+def _maybe_adjust_config(config):
+    rank_pattern = config["rank_pattern"].copy()
+    target_modules = config["target_modules"]
+    original_r = config["r"]
+
+    for key in list(rank_pattern.keys()):
+        key_rank = rank_pattern[key]
+
+        # try to detect ambiguity
+        exact_matches = [mod for mod in target_modules if mod == key]
```

Comment:
Not sure if this can happen here, but `target_modules` can also be a `str`, in which case this loop would loop over the chars of the `str`. The technically correct way to match LoRA keys in PEFT is to use `LoraModel._check_target_module_exists(lora_config, key)`. But maybe in this context this is not possible and we can ignore it? (in that case, maybe add a comment that says so)

Reply:
I have added a comment but how can we get access to `LoraModel. _check_target_module_exists()` in this context?

Reply:
`LoraModel._check_target_module_exists()` is a static method, so importing `peft.tuners.lora.LoraModel` is enough to access it, no need to initialize anything.

Reply:
@BenjaminBossan I thought about it. I think it's better not rely on a private method here for now (I remember there was an issue long time back when we did this). Happy to also add a comment denoting that we should move to using a public method from `peft` when it becomes available. 

But to make this more robust, maybe I could check if `target_modules` is a `List` and if not, we make it one? 

WDYT?

Reply:
If you don't want to use the private method, you can use `from peft.tuners.tuners_utils import check_target_module_exists`, which does the same thing.

> maybe I could check if `target_modules` is a `List` and if not, we make it one?

This would not work if the `target_modules` variable is a str, as this requires a different logic to resolve the name.

Reply:
Sorry for making this more complicated than necessary. 

> Sorry, I don't understand this comment.

I mean to add the following to the method (`_maybe_adjust_config()`) at the beginning:

```diff
+ if not isinstance(target_modules, (list, set)):
+    target_modules = [target_modules]
```

Is this clear now?

Reply:
No, this wouldn't work as intended, because if `target_modules` is a str, it could be something like `lin.*` so `[mod for mod in target_modules if mod == key]` won't match anything. In sum, let's not spend too much time on this edge case. How about just raising when it's a str?

Reply:
Fair. I will do that. Thanks for explaining.